scripts/globals/barge.lua

```diff
@@ -0,0 +1,266 @@
+        return
+    elseif aboard == 1 then -- South Landing
+        departEvent = 14
+        kickEvent = 33
+    elseif aboard == 2 then -- Central Landing
+        departEvent = 40
+        kickEvent = 42
+    elseif aboard == 3 then -- North Landing
+        departEvent = 16
+        kickEvent = 34
+    end
+
+    if player:hasKeyItem(xi.ki.BARGE_TICKET) then
+        player:messageSpecial(ID.text.BARGE_TICKET_USED, 0, xi.ki.BARGE_TICKET)
+        player:delKeyItem(xi.ki.BARGE_TICKET)
+        player:startEvent(departEvent)
+    elseif player:hasKeyItem(xi.ki.BARGE_MULTI_TICKET) then
+        local uses = player:getCharVar('[barge]multiTicket')
+
+        if uses <= 1 then
+            player:messageSpecial(ID.text.BARGE_TICKET_USED, 0, xi.ki.BARGE_MULTI_TICKET)
+            player:delKeyItem(xi.ki.BARGE_MULTI_TICKET)
+        else
+            player:messageSpecial(ID.text.BARGE_TICKET_REMAINING, 0, xi.ki.BARGE_MULTI_TICKET, uses - 1)
+        end
+
+        player:setCharVar('[barge]multiTicket', uses - 1)
+        player:startEvent(departEvent)
+    else
+        player:startEvent(kickEvent)
+    end
+end
+
+xi.barge.onTicketShopTrigger = function(player, eventId)
+    player:setCharVar('[barge]currentTicket', 0) -- Set ticket to 0 in case something breaks
+
+    if
+        player:hasKeyItem(xi.ki.BARGE_TICKET) and
+        player:hasKeyItem(xi.ki.BARGE_MULTI_TICKET)
+    then
+        player:setCharVar('[barge]currentTicket', 3)
+    elseif player:hasKeyItem(xi.ki.BARGE_MULTI_TICKET) then
+        player:setCharVar('[barge]currentTicket', 2)
+    elseif player:hasKeyItem(xi.ki.BARGE_TICKET) then
+        player:setCharVar('[barge]currentTicket', 1)
+    else
+        player:setCharVar('[barge]currentTicket', 0)
+    end
+
+    -- Params (KI1, KI2, Price of KI1, Price of KI2, Multiticket #s left, Which tickets does player have)
+    player:startEvent(eventId, xi.ki.BARGE_TICKET, xi.ki.BARGE_MULTI_TICKET , 50, 300, player:getCharVar('[barge]multiTicket'), player:getCharVar('[barge]currentTicket'), 1, 4095) -- Start event
+end
+
+xi.barge.onTicketShopEventFinish = function(player, csid, option)
+    local ID = zones[player:getZoneID()]
+    local currentTicket = player:getCharVar('[barge]currentTicket')
+    local numberTicket = player:getCharVar('[barge]multiTicket')
+
+    -- Option 1: BARGE_TICKET
+    -- Option 2: MULTI_TICKET
+    if
+        csid == 31 or
+        csid == 32 or
+        csid == 43
+    then
+        local gil = player:getGil()
+        if option == 1 and (currentTicket == 1 or currentTicket == 3) then -- If you have BARGE_TICKET
+            -- Event auto plays the correct message
+        elseif option == 1 then
+            if gil >= 50 then
+                player:delGil(50)
+                player:addKeyItem(xi.ki.BARGE_TICKET)
+                player:messageSpecial(ID.text.KEYITEM_OBTAINED, xi.ki.BARGE_TICKET)
+            else
+                player:messageSpecial(7440) -- too poor
```

Comment:
The proper message when you don't have enough gil is this message but I couldn't get the NPC to respond with it in the event so this currently looks like:

![image](https://github.com/LandSandBoat/server/assets/166072302/f417477e-d4be-45f6-87e2-a0678f525642)

But the NPC should only respond with the too poor message. It should also come from the NPC similar to the Riche and Eunirange messages.

Reply:
Was able to get the NPC to say it using `player:showText()` but you still get the thank you message before the too poor message.

Reply:
I am not at a computer to check, but it could be due to default actions being called because the onTrigger isn't starting an event